riverdriver/riverpgxv5/internal/dbsqlc/river_job.sql

```diff
@@ -276,9 +278,43 @@ INSERT INTO river_job(
     @queue,
     coalesce(sqlc.narg('scheduled_at')::timestamptz, now()),
     @state,
-    coalesce(@tags::varchar(255)[], '{}')
+    coalesce(@tags::varchar(255)[], '{}'),
+    @unique_key
 ) RETURNING *;
 
+-- name: JobInsertUnique :one
```

Comment:
This op is ~the same as `JobInsertFast` except for the `ON CONFLICT`.

We may want to consider just rollling the two up together so that every job insert is just an upsert to reduce on the number of separate ops in here we have.

I left them separate for now because I wrote a benchmark and the upsert version regularly seems to come out as a couple percent slower:

```
$ go test -bench=BenchmarkDriverRiverPgxV5Insert .
goos: darwin
goarch: arm64
pkg: github.com/riverqueue/river
BenchmarkDriverRiverPgxV5Insert/InsertFast-8               12199             97738 ns/op
BenchmarkDriverRiverPgxV5Insert/InsertUnique-8             10000            100346 ns/op
PASS
ok      github.com/riverqueue/river     15.302s
```

It's not that big of a difference though, so an argument could be made that long term the extra operations aren't worth it.

Reply:
Actually, those bench results show quite a large different. But depending on the run, some are much closer:

```
$ go test -bench=BenchmarkDriverRiverPgxV5Insert .
goos: darwin
goarch: arm64
pkg: github.com/riverqueue/river
BenchmarkDriverRiverPgxV5Insert/InsertFast-8               12384            102295 ns/op
BenchmarkDriverRiverPgxV5Insert/InsertUnique-8             12135            112301 ns/op
PASS
ok      github.com/riverqueue/river     18.364s
```

Most runs tend to be a closer to the first than the second, so probably good reason to keep them as separate ops for now.